applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_implicit_dynamic_solver.py

```diff
@@ -58,12 +78,58 @@ def AddDofs(self):
         self._add_dynamic_dofs()
         self.print_on_rank_zero("::[ImplicitMechanicalSolver]:: ", "DOF's ADDED")
 
+    def InitializeSolutionStep(self):
+        # Using the base InitializeSolutionStep
+        super(ImplicitMechanicalSolver, self).InitializeSolutionStep()
+
+        # We compute for the different parts (each body can have it own damping)
+        if self.dynamic_settings["determine_rayleigh_damping"].GetBool() and self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] == 1:
+            import KratosMultiphysics.kratos_utilities as kratos_utils
+            if kratos_utils.IsApplicationAvailable("ExternalSolversApplication"):
+                from KratosMultiphysics import ExternalSolversApplication
+            else:
+                raise Exception("ExternalSolversApplication not available")
+
+            # The general damping ratios
+            damping_ratio_0 = self.dynamic_settings["determine_rayleigh_damping_settings"]["damping_ratio_0"].GetDouble()
+            damping_ratio_1 = self.dynamic_settings["determine_rayleigh_damping_settings"]["damping_ratio_1"].GetDouble()
+
+            # We get the model parts which divide the problem
+            structural_parts = self.__extract_model_parts(self.settings["problem_domain_sub_model_part_list"])
+            for part in structural_parts:
+
+                # Reseting values
+                self.dynamic_settings["determine_rayleigh_damping_settings"]["damping_ratio_0"].SetDouble(damping_ratio_0)
+                self.dynamic_settings["determine_rayleigh_damping_settings"]["damping_ratio_1"].SetDouble(damping_ratio_1)
+
+                # We check if the values are previously defined
+                properties = part.GetProperties()
+                for prop in properties:
+                    if prop.Has(StructuralMechanicsApplication.SYSTEM_DAMPING_RATIO):
+                        self.dynamic_settings["determine_rayleigh_damping_settings"]["damping_ratio_0"].SetDouble(prop.GetValue(StructuralMechanicsApplication.SYSTEM_DAMPING_RATIO))
+                        break
+                for prop in properties:
+                    if prop.Has(StructuralMechanicsApplication.SECOND_SYSTEM_DAMPING_RATIO):
+                        self.dynamic_settings["determine_rayleigh_damping_settings"]["damping_ratio_1"].SetDouble(prop.GetValue(StructuralMechanicsApplication.SECOND_SYSTEM_DAMPING_RATIO))
+                        break
```

Comment:
This does not work if I want different damping_ratios on different parts of the structure, which is a valid case if one uses different materials

=> you could define a global one and but if sth is specified on subpart it should be used

Reply:
This is precisely for that purpose